p2p/transport/websocket/websocket.go

```diff
@@ -179,7 +173,26 @@ func (t *WebsocketTransport) maDial(ctx context.Context, raddr ma.Multiaddr) (ma
 		return nil, err
 	}
 	isWss := wsurl.Scheme == "wss"
-	dialer := ws.Dialer{HandshakeTimeout: 30 * time.Second}
+	wsurlCopy := *wsurl
+	remoteAddr := addrWrapper{URL: &wsurlCopy}
+	localAddrChan := make(chan addrWrapper, 1)
+
+	transport := &http.Transport{
+		DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
Why are we setting the `DialContext` here and below (in the `isWss` branch)?

Reply:
To fill the `localAddr` into localAddrChan in case `!isWss`. This is the hackiest part of this. It doesn't seem like the websocket library lets us see the localAddr for the connection (which we need to fulfill the `net.Conn` interface). We can get it as part of the underlying dial, which is what I'm doing here.

Reply:
Can we a `!isWss` code path and set it there? That would make the code slightly easier to understand.

Reply:
It's harder to understand imo: https://gist.github.com/MarcoPolo/ddb3c159d1358d6e261a78282425839c